library/blob.c

```diff
@@ -330,10 +330,12 @@ inline static uint64_t eblob_validate_ctl_flags(struct eblob_backend *b, uint64_
 	if (b->cfg.blob_flags & EBLOB_NO_FOOTER)
 		flags |= BLOB_DISK_CTL_NOCSUM;
 
-	if (!(flags & BLOB_DISK_CTL_NOCSUM))
-		flags |= BLOB_DISK_CTL_CHUNKED_CSUM;
-	else
-		flags &= ~BLOB_DISK_CTL_CHUNKED_CSUM;
+	/*
+	 * We have to add BLOB_DISK_CTL_CHUNKED_CSUM because it shows
+	 * that footer was prepared with specific size, even if the record
+	 * has BLOB_DISK_CTL_NOCSUM flag.
+	 */
+	flags |= BLOB_DISK_CTL_CHUNKED_CSUM;
```

Comment:
I meant different thing.
Let's suppose we have a record with BLOB_DISK_CTL_CHUNKED_CSUM bit set, it also has a footer, but it is empty.
Previously it was enough just to look at record's flag to determine that there is no checksum and thus eblob should not try to verify it. We plan to always have BLOB_DISK_CTL_CHUNKED_CSUM bit now, and how should eblob make verification decision? Looks like it will have to go to the footer, calculate its size, determine that it is empty, and only then decide not to verify checksum.

What's wrong with *not* setting BLOB_DISK_CTL_CHUNKED_CSUM bit if we do not actually plan to put checksum into footer?